mne/forward/forward.py

```diff
@@ -1495,14 +1501,14 @@ def restrict_forward_to_stc(fwd, stc):
     if fwd['sol_grad'] is not None:
         fwd_out['_orig_sol_grad'] = fwd['_orig_sol_grad'][:, idx_grad]
 
-    for i in range(2):
-        fwd_out['src'][i]['vertno'] = stc.vertices[i]
-        fwd_out['src'][i]['nuse'] = len(stc.vertices[i])
-        fwd_out['src'][i]['inuse'] = fwd['src'][i]['inuse'].copy()
-        fwd_out['src'][i]['inuse'].fill(0)
-        fwd_out['src'][i]['inuse'][stc.vertices[i]] = 1
-        fwd_out['src'][i]['use_tris'] = np.array([[]], int)
-        fwd_out['src'][i]['nuse_tri'] = np.array([0])
+    for si, s in enumerate(fwd_out['src']):
+        vertno = idx_sel[1][idx_sel[0] == si]
+        s['inuse'].fill(0)
+        s['inuse'][vertno] = 1
+        s['nuse'] = len(vertno)
+        s['vertno'] = vertno
+        s['use_tris'] = np.array([[]], int)
+        s['nuse_tri'] = np.array([0])
 
     return fwd_out
 
```

Comment:
For the fwd functions, it might be good if someone more familiar with the module could have a second look :)